tests/integration_tests/functional/test_cpu_features_aarch64.py

```diff
@@ -3,83 +3,63 @@
-)
+G3_SVE_AND_PAC = set("paca pacg sve svebf16 svei8mm".split())
 
-DEFAULT_G3_FEATURES_V1N1 = DEFAULT_G2_FEATURES
 
+def test_guest_cpu_features(uvm_any):
+    """Check the CPU features for a microvm with different CPU templates"""
 
-def _check_cpu_features_arm(test_microvm, guest_kv, template_name=None):
-    expected_cpu_features = {"Flags": []}
-    match cpuid_utils.get_cpu_model_name(), guest_kv, template_name:
-        case CpuModel.ARM_NEOVERSE_N1, _, "v1n1":
-            expected_cpu_features = DEFAULT_G2_FEATURES
-        case CpuModel.ARM_NEOVERSE_N1, _, None:
-            expected_cpu_features = DEFAULT_G2_FEATURES
+    vm = uvm_any
+    expected_cpu_features = set()
+    match global_props.cpu_model, vm.cpu_template_name:
+        case CpuModel.ARM_NEOVERSE_N1, "v1n1":
+            expected_cpu_features = G2_FEATS
+        case CpuModel.ARM_NEOVERSE_N1, None:
+            expected_cpu_features = G2_FEATS
 
         # [cm]7g with guest kernel 5.10 and later
-        case CpuModel.ARM_NEOVERSE_V1, _, "v1n1":
-            expected_cpu_features = DEFAULT_G3_FEATURES_V1N1
-        case CpuModel.ARM_NEOVERSE_V1, _, "aarch64_with_sve_and_pac":
-            expected_cpu_features = DEFAULT_G3_FEATURES_WITH_SVE_AND_PAC_5_10
-        case CpuModel.ARM_NEOVERSE_V1, _, None:
-            expected_cpu_features = DEFAULT_G3_FEATURES_5_10
-
-    _, stdout, _ = test_microvm.ssh.check_output(CPU_FEATURES_CMD)
-    flags = set(stdout.strip().split(" "))
-    assert flags == expected_cpu_features
+        case CpuModel.ARM_NEOVERSE_V1, "v1n1":
+            expected_cpu_features = G2_FEATS
+        case CpuModel.ARM_NEOVERSE_V1, "aarch64_with_sve_and_pac":
+            expected_cpu_features = G3_FEATS | G3_SVE_AND_PAC
+        case CpuModel.ARM_NEOVERSE_V1, None:
+            expected_cpu_features = G3_FEATS
 
+    guest_feats = set(vm.ssh.check_output(CPU_FEATURES_CMD).stdout.split())
+    assert guest_feats == expected_cpu_features
 
-def get_cpu_template_dir(cpu_template):
-    """
-    Utility function to return a valid string which will be used as
-    name of the directory where snapshot artifacts are stored during
-    snapshot test and loaded from during restore test.
 
-    """
-    return cpu_template if cpu_template else "none"
-
-
-@pytest.mark.skipif(
-    PLATFORM != "aarch64",
-    reason="This is aarch64 specific test.",
-)
-def test_host_vs_guest_cpu_features_aarch64(uvm_nano):
+def test_host_vs_guest_cpu_features(uvm_nano):
     """Check CPU features host vs guest"""
 
     vm = uvm_nano
     vm.add_net_iface()
     vm.start()
-    host_feats = set(utils.check_output(CPU_FEATURES_CMD).stdout.strip().split(" "))
-    guest_feats = set(vm.ssh.check_output(CPU_FEATURES_CMD).stdout.strip().split(" "))
-
-    cpu_model = cpuid_utils.get_cpu_model_name()
+    host_feats = parse_cpu_flags(utils.check_output(GET_CPU_FLAGS_CMD).stdout)
+    guest_feats = parse_cpu_flags(vm.ssh.check_output(GET_CPU_FLAGS_CMD).stdout)
```

Comment:
`parse_cpu_flags()` not defined anywhere but will be removed in a subsequent commit. Is it used for temporary purpose?

Reply:
Hmm I think there was a rebase gone wrong here.